client/src/components/authentication/SignInForm.js

```diff
@@ -29,29 +29,32 @@ const SignInForm = () => {
     setErrors(newErrors);
   };
 
-  const onSubmit = (event) => {
+  const onSubmit = async (event) => {
     event.preventDefault();
-    validateInput(userPayload);
-    if (Object.keys(errors).length === 0) {
-      fetch("/api/v1/user-sessions", {
-        method: "post",
-        body: JSON.stringify(userPayload),
-        headers: new Headers({
-          "Content-Type": "application/json",
-        }),
-      }).then((resp) => {
-        if (resp.ok) {
-          resp.json().then(() => {
-            setShouldRedirect(true);
-          });
-        } else {
+    // validateInput(userPayload);
+    try {
+      if (Object.keys(errors).length === 0) {
+        const response = await fetch("/api/v1/user-sessions", {
+          method: "post",
+          body: JSON.stringify(userPayload),
+          headers: new Headers({
+            "Content-Type": "application/json",
+          }),
+        })
+        
+        if (!response.ok) {
           const errorMessage = `${resp.status} (${resp.statusText})`;
-          const error = new Error(errorMessage);
-          throw error;
-        }
-      });
+          throw new Error(errorMessage);
+        } 
+        
+        const userData = await response.json()
```

Comment:
Currently we aren't making use of this `useData`, so parsing it from the response is an extra step that can slow down our application. However, we can make use of this before we redirect to ensure we have the correct data. After we parse the response, we can conditionally check if userData contains something like `user`, _then_ we want to redirect